Makefile

```diff
@@ -968,14 +986,14 @@ $(TARGET_ELF):  $(TARGET_OBJS)
 # Compile
 $(OBJECT_DIR)/$(TARGET)/%.o: %.c
 	$(V1) mkdir -p $(dir $@)
-	$(V1) $(if $(findstring $(subst ./src/main/,,$<), $(SPEED_OPTIMISED_SRC)), \
+	$(V1) $(if $(findstring $(notdir $<),$(SPEED_OPTIMISED_SRC)), \
```

Comment:
OK, you have a go at it. 
The problem I have when searching the file lists seems to some files are entered with path while some are plain file name. The later then relies on VPATH to be found.  Example:

    USBCDC_SRC  = $(notdir $(wildcard $(USBCDC_DIR)/src/*.c))

The only generic way was to do $(notdir) also..... I was thinking of an entirely different approach, not using any searches in filelist, but rather separate specific compile rules for each opt group of files. I did some experiments on that, but dropped it. May have to look into that instead. But first lets see what you got up your sleeves.... Parking this PR for now.